vault/identity_store_oidc.go

```diff
@@ -1344,35 +1470,79 @@ func (i *IdentityStore) oidcKeyRotation(ctx context.Context, s logical.Storage)
 
 // oidcPeriodFunc is invoked by the backend's periodFunc and runs regular key
 // rotations and expiration actions.
-func (i *IdentityStore) oidcPeriodicFunc(ctx context.Context, s logical.Storage) {
-	nextRun := time.Time{}
+func (i *IdentityStore) oidcPeriodicFunc(ctx context.Context) {
+	var nextRun time.Time
+	now := time.Now()
 
-	if v, ok := i.oidcCache.Get("nextRun"); ok {
+	nsPaths := i.listNamespacePaths(ctx)
+
+	if v, ok := i.oidcCache.Get(nil, "nextRun"); ok {
 		nextRun = v.(time.Time)
 	}
 
 	// The condition here is for performance, not precise timing. The actions can
 	// be run at any time safely, but there is no need to invoke them (which
 	// might be somewhat expensive if there are many roles/keys) if we're not
 	// past any rotation/expiration TTLs.
-	if time.Now().After(nextRun) {
-		nextRotation, err := i.oidcKeyRotation(ctx, s)
-		if err != nil {
-			i.Logger().Warn("error rotating OIDC keys", "err", err)
-		}
+	if now.After(nextRun) {
+		// Initialize to a fairly distant next run time. This will be brought in
+		// based on key rotation times.
+		nextRun = now.Add(24 * time.Hour)
 
-		nextExpiration, err := i.expireOIDCPublicKeys(ctx, s)
-		if err != nil {
-			i.Logger().Warn("error expiring OIDC public keys", "err", err)
-		}
+		for _, nsPath := range nsPaths {
```

Comment:
I think that the logic in this range also needs to happen in the nil namespace, which, I don't think it is at the moment.

Reply:
The cache calls here do use `nilNamespace`.